stetho/src/main/java/com/facebook/stetho/inspector/helper/ThreadBoundProxy.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.helper;
+
+import com.facebook.stetho.common.ThreadBound;
+import com.facebook.stetho.common.UncheckedCallable;
+import com.facebook.stetho.common.Util;
+
+/*
+This class is for those cases when is a class' threading
+policy is determined by one of its member variables.
+ */
+public abstract class ThreadBoundProxy implements ThreadBound {
+  private ThreadBound mEnforcer;
+
+  public ThreadBoundProxy(ThreadBound enforcer) {
+    mEnforcer = Util.throwIfNull(enforcer);
+  }
+
+  @Override
+  public boolean checkThreadAccess() {
```

Comment:
let's mark all of the `ThreadBound` methods as `final`, so long as nobody else is overriding them. The whole point of deriving from this class is to allow the base class to handle them, so why not make sure derived classes can't even accidentally override them :)